File: app/gui/design/tokens/spacing.py
"""
SOC-IQ Design System
Spacing Design Tokens

Defines the standard spacing scale used throughout the application.

All margins, padding and layout spacing should reference these
tokens instead of hard-coded pixel values.
"""

from __future__ import annotations


class Spacing:
    """Semantic spacing tokens (values in pixels)."""

    # Micro spacing
    NONE = 0
    XXS = 2
    XS = 4

    # Standard spacing
    SM = 8
    MD = 12
    LG = 16

    # Large spacing
    XL = 24
    XXL = 32
    XXXL = 48

    # Page layout
    PAGE_MARGIN = 24
    SECTION_GAP = 24
    CARD_PADDING = 16
    PANEL_PADDING = 20

    # Dense data views
    TABLE_CELL_PADDING = 8
    ROW_HEIGHT = 32

    # Forms
    LABEL_GAP = 6
    FIELD_GAP = 12

    # Sidebar
    SIDEBAR_PADDING = 16
    SIDEBAR_ITEM_HEIGHT = 44
    SIDEBAR_ICON_GAP = 12

    # Toolbar
    TOOLBAR_HEIGHT = 52

    # Header
    HEADER_HEIGHT = 64

    # Status bar
    STATUSBAR_HEIGHT = 28

File: frontend/src/styles/tokens/spacing.ts
/**
 * SOC-IQ Design System — Spacing Tokens
 * Ported 1:1 from `app/gui/design/tokens/spacing.py` (values in px).
 */

export const Spacing = {
  none: 0,
  xxs: 2,
  xs: 4,

  sm: 8,
  md: 12,
  lg: 16,

  xl: 24,
  xxl: 32,
  xxxl: 48,

  pageMargin: 24,
  sectionGap: 24,
  cardPadding: 16,
  panelPadding: 20,

  gridGap: 20,
  cardGap: 16,
  widgetGap: 12,

  tableCellPadding: 8,
  rowHeight: 32,

  labelGap: 6,
  fieldGap: 12,
  contentGap: 16,

  sidebarPadding: 16,
  sidebarItemHeight: 44,
  sidebarIconGap: 12,

  toolbarHeight: 52,
  headerHeight: 64,
  statusbarHeight: 28,

  dialogPadding: 24,
} as const;

export type SpacingToken = typeof Spacing;
